src/scripts/main.js

```diff
@@ -1,3 +1,327 @@
+      if (num === array[index + 1]) {
+        array[index + 1] = null;
+        scoring(num * 2);
+
+        return num * 2;
+      } else {
+        return num;
+      }
+    }).filter((num) => num);
+
+    for (let i = 0; i < row.cells.length; i++) {
+      const cell = row.cells[i];
+      const newValue = rowValuesMerged.shift();
+
+      if (newValue) {
+        cell.classList.add(`field-cell--${newValue}`);
+        cell.textContent = newValue;
+      }
+    }
+  });
+
+  gameOver();
+}
+
+function moveRowsRight(rows) {
+  rows.forEach(row => {
+    const rowValues = [];
+
+    for (let i = row.cells.length - 1; i >= 0; i--) {
+      const cell = row.cells[i];
+
+      if (cell.textContent !== '') {
+        const value = cell.textContent;
+
+        cell.classList.remove(`field-cell--${value}`);
+        rowValues.unshift(value);
+        cell.textContent = '';
+      }
+    }
+
+    const rowValuesMerged = rowValues.map((num, index, array) => {
+      if (num === array[index + 1]) {
+        array[index + 1] = null;
+        scoring(num * 2);
+
+        return num * 2;
+      } else {
+        return num;
+      }
+    }).filter((num) => num);
+
+    for (let i = row.cells.length - 1; i >= 0; i--) {
+      const cell = row.cells[i];
+      const newValue = rowValuesMerged.pop();
+
+      if (newValue) {
+        cell.classList.add(`field-cell--${newValue}`);
+        cell.textContent = newValue;
+      }
+    }
+  });
+
+  gameOver();
+}
+
+function moveColumnsUp(columns) {
+  columns.forEach(column => {
+    const columnValues = [...column.map(el => el.textContent)];
+
+    const columnValuesMerged = columnValues.map((num, index, array) => {
+      if (num === array[index + 1]) {
+        array[index + 1] = null;
+        scoring(num * 2);
+
+        return num * 2;
+      } else {
+        return num;
+      }
+    }).filter((num) => num);
+
+    for (let i = 0; i < column.length; i++) {
+      const cell = column[i];
+
+      cell.classList.remove(`field-cell--${cell.textContent}`);
+
+      const newValue = columnValuesMerged.shift();
+
+      cell.textContent = newValue;
+      cell.classList.add(`field-cell--${newValue}`);
+    }
+  });
+
+  gameOver();
+}
+
+function moveColumnsDown(columns) {
+  columns.forEach(column => {
+    const columnValues = [...column.map(el => el.textContent)].reverse();
+
+    const columnValuesMerged = columnValues.map((num, index, array) => {
+      if (num === array[index + 1]) {
+        array[index + 1] = null;
+        scoring(num * 2);
+
+        return num * 2;
+      } else {
+        return num;
+      }
+    }).filter((num) => num);
+
+    for (let i = column.length - 1; i >= 0; i--) {
+      const cell = column[i];
+
+      cell.classList.remove(`field-cell--${cell.textContent}`);
+
+      const newValue = columnValuesMerged.shift();
+
+      cell.textContent = newValue;
+      cell.classList.add(`field-cell--${newValue}`);
+    }
```

Comment:
Could you try to generalize moving logic and avoid code duplication? 
You make the same things inside every moving function and can make a general method that will take the params it needs. So, potentially your code will be shorter, cleaner, and more readable.

Reply:
Fixed all.